src/hooks/http.test.js

```diff
@@ -493,6 +502,157 @@ describe('http hook', () => {
     );
   });
 
+  test('httpRequestWrapper - simple flow', () => {
+    TracerGlobals.setHandlerInputs({
+      event: {},
+      context: {
+        awsRequestId: HttpSpanBuilder.DEFAULT_PARENT_ID,
+        invokedFunctionArn: HttpSpanBuilder.DEFAULT_ARN,
+        functionVersion: HttpSpanBuilder.DEFAULT_VERSION,
+      },
+    });
+    const requestData = HttpSpanBuilder.DEFAULT_REQUEST_DATA;
+    const responseData = {
+      statusCode: 200,
+      body: 'OK',
+    };
+
+    HttpsScenarioBuilder.appendNextResponse(responseData.body);
+    const wrappedRequest = httpHook.httpRequestWrapper(HttpsMocker.request);
+
+    wrappedRequest(requestData, () => {});
+
+    const spans = SpansContainer.getSpans();
+
+    const expectedSpan = new HttpSpanBuilder()
+      .withStarted(spans[0].started)
+      .withEnded(spans[0].ended)
+      .withSpanId(spans[0].id)
+      .withHttpInfo({
+        host: HttpSpanBuilder.DEFAULT_HOST,
+        request: requestData,
+        response: responseData,
+      })
+      .withRequestTimesFromSpan(spans[0])
+      .build();
+
+    expect(spans).toEqual([expectedSpan]);
+  });
+
+  test('httpRequestWrapper - added span before request finish', () => {
+    utils.setTimeoutTimerEnabled();
+    TracerGlobals.setHandlerInputs({
+      event: {},
+      context: {
+        awsRequestId: HttpSpanBuilder.DEFAULT_PARENT_ID,
+        invokedFunctionArn: HttpSpanBuilder.DEFAULT_ARN,
+        functionVersion: HttpSpanBuilder.DEFAULT_VERSION,
+      },
```

Comment:
Maybe extract this to a variable or a function, it is used in more than one test.

Reply:
Yes you right, added a builder for that. tell me what you think